### PR TITLE
common: add sim-on-hardware information

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: common/source/docs/common-sim-on-hardware.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+-  only quadcopters have been tested as working
+
+--------------------
+Obtaining a Firmware
+--------------------
+
+Using a pre-compiled binary
+...........................
+
+We provide a pre-compiled binary for CubeOrange at :ref:`our firmware server <https://firmware.ardupilot.org>`.
```
Comment (Hwurzburg) [inline common/source/docs/common-sim-on-hardware.rst:34]: cant find it

File: common/source/docs/common-sim-on-hardware.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+Simulation on Hardware allows:
+
+-  checking that mission structure is correct
+-  control surfaces move appropriately throughout a simulated mission
+-  ensuring that physical failsafes (e.g. parachutes) function appropriately without putting a vehicle at risk
+-  communications infrastructure works in the face of real traffic
+
+.. note::
+
+   Simulation on Hardware is being improved all the time.  Do not be surprised if it is suggested you run the master branch to be able to take advantage of new features that have been added.
```
Comment (Hwurzburg) [inline common/source/docs/common-sim-on-hardware.rst:16]:
```suggestion
   Simulation on Hardware is being improved all the time.  It is suggested you run the master branch ("latest" firmware) to be able to take advantage of any new features that have been added.
```

File: common/source/docs/common-sim-on-hardware.rst [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+-----------
+Limitations
+-----------
+
+-  actuator output is not supported without recompilation.  This is currently considered a safety feature
+-  parameter space is shared between the real aircraft and the simulated aircraft.  This leads to problems with sensor calibration as the sensor suite differs between the real and virtual aircraft.  Thus parameters must be wiped and rewritten when moving between real and simulating firmwares.
```
Comment (Hwurzburg) [inline common/source/docs/common-sim-on-hardware.rst:23]:
```suggestion
-  parameter space is shared between the real aircraft and the simulated aircraft.  This leads to problems with sensor calibration as the sensor suite differs between the real and virtual aircraft.  Thus parameters must be wiped and rewritten to the vehicle when moving between real and simulation  firmware.
```

File: common/source/docs/common-sim-on-hardware.rst [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+-  while it is possible to run on flash-constrained (1MB) boards, many other features will need to be compiled out to allow it to fit
+-  only quadcopters have been tested as working
```
Comment (Hwurzburg) [inline common/source/docs/common-sim-on-hardware.rst:25]:
```suggestion
-  only Quadcopters have been tested as working
```

File: common/source/docs/common-sim-on-hardware.rst [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+.. note::
+
+   Simulation on Hardware is being improved all the time.  Do not be surprised if it is suggested you run the master branch to be able to take advantage of new features that have been added.
+
+-----------
+Limitations
+-----------
+
+-  actuator output is not supported without recompilation.  This is currently considered a safety feature
```
Comment (Hwurzburg) [inline common/source/docs/common-sim-on-hardware.rst:22]:
```suggestion
-  actuator output is not supported without recompilation of the simulation firmware as posted on the :ref:`firmware server <https://firmware.ardupilot.org>` .  This is currently considered a safety feature
```

File: common/source/docs/common-sim-on-hardware.rst [hunk 6]
```diff
@@ -0,0 +1,70 @@
+.. _common-sim-on-hardware:
```
Comment (Hwurzburg) [inline common/source/docs/common-sim-on-hardware.rst:1]: this should be a dev topic I would think...not a vehicle common topic

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
I take it this is WIP and have marked it as such...
It needs:
- better explanation of system at top, other than features...what the heck is it??
- setup and connection information
- TOC entry in the SIM section of dev

**Comment by nkruzan** [comment]
Just some notes: 
PR for initial support in main code base is here https://github.com/ArduPilot/ardupilot/pull/19810
PR for CubeOrange-SimOnHardware is here https://github.com/ArduPilot/ardupilot/pull/20339

**Review by peterbarker — COMMENTED** [review]
>    better explanation of system at top, other than features...what the heck is it??

Added Some Words

>    setup and connection information

It really should be as simple as what's covered in here, but I'm expecting to add more notes as I receive feedback.

>    TOC entry in the SIM section of dev

I've added a note on a simulation page - but I'm pretty sure that's not what you meant here.  Can you clarify, please?

File: common/source/docs/common-sim-on-hardware.rst (hunk 1 above)
Comment (peterbarker) [reply common/source/docs/common-sim-on-hardware.rst]: Fixed.

**Comment by peterbarker** [comment]
@Hwurzburg what do you think this still needs before merge?

**Comment by Hwurzburg** [comment]
@peterbarker   made some changes, if okay I will merge

File: dev/source/docs/sim-on-hardware.rst [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+======================
+
+ArduPilot's Simulation on Hardware is similar to features sometimes known as "Hardware in the Loop".  Contrary to many of these systems, however, the simulation is not run externally (with the resulting environmental data passed to the vehicle), rather the vehicle's own autopilot runs both the flight control software and the simulation.  ArduPilot's native SITL simulation environment is re-used for this purpose.
+
+Simulation on Hardware allows:
+
+- checking that mission structure is correct
+- control surfaces move appropriately throughout a simulated mission
+- ensuring that physical failsafes (e.g. parachutes) function appropriately without putting a vehicle at risk
+- communications infrastructure works in the face of real traffic
```
Comment (peterbarker) [inline dev/source/docs/sim-on-hardware.rst:14]:
```suggestion
Simulation on Hardware allows checking that:

- mission structure is correct
- control surfaces move appropriately throughout a simulated mission
- physical failsafes (e.g. parachutes) function appropriately without putting a vehicle at risk
- communications infrastructure works in the face of real traffic
```

File: dev/source/docs/sim-on-hardware.rst [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+
+.. note::
+
+   Simulation on Hardware is being improved all the time.  It is suggested you run the master branch ("latest" firmware) to be able to take advantage of any new features that have been added.
+
+The firmware is loaded on the autopilot and a GCS attached. The GPS initial location data is provided by the simulation and can be changed when compiling the firware. Setup, arming, and flying the simulation occurs as if sim_vehicle.py and the associated physics models for the vehicle and frame was used.
```
Comment (peterbarker) [inline dev/source/docs/sim-on-hardware.rst:20]:
```suggestion
The firmware is loaded on the autopilot and a GCS attached. The GPS initial location data is provided by the simulation and can be changed with `SIM_OPOS_LAT`, `SIM_OPOS_LNG`, `SIM_OPOS_ALT` and `SIM_OPOS_HDG` parameters. Setup, arming, and flying the simulation occurs as if sim_vehicle.py and the associated physics models for the vehicle and frame was used.
```

File: dev/source/docs/sim-on-hardware.rst [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+
+Limitations
+===========
+
+-  Actuator outputs (motor and servo) are not supported in the firmware as posted on the :ref:`firmware server <https://firmware.ardupilot.org>` without recompilation.  This is currently considered a safety feature.
+
+.. warning:: relay outputs currently are not prevented from operation. Be careful if they are being used for payload releases or parachute deployment.
```
Comment (peterbarker) [inline dev/source/docs/sim-on-hardware.rst:27]: You've fixed this now.

File: dev/source/docs/sim-on-hardware.rst [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+
+-  Parameter space is shared between the real aircraft and the simulated aircraft.  This leads to problems with sensor calibration as the sensor suite differs between the real and virtual aircraft.  Thus parameters must be wiped and rewritten to the vehicle when moving between real and simulation firmware.
+- While it is possible to run on flash-constrained (1MB) boards, many other features will need to be compiled out to allow it to fit.
+- Only Quadcopters have been tested as working.
```
Comment (peterbarker) [inline dev/source/docs/sim-on-hardware.rst:31]:
```suggestion
```

Others have tested planes.  They were keen.

File: dev/source/docs/sim-on-hardware.rst [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+    set -e
+    set -x
+
+    BOARD=MatekH743
+    THISDIR=$(dirname $0)
+
+    ./waf configure \
+      --board=$BOARD \
+      --extra-hwdef="$THISDIR/extra-hwdef-sitl-on-hw.dat" \  a typical file is shown `here <https://github.com/ArduPilot/ardupilot/blob/master/libraries/SITL/examples/on-hardware/extra-hwdef-sitl-on-hw.dat>`__
+      --default-param="$THISDIR/default.param" \ a typical default param file is shown `here <https://github.com/ArduPilot/ardupilot/blob/master/libraries/SITL/examples/on-hardware/default.param>`__
```
Comment (peterbarker) [inline dev/source/docs/sim-on-hardware.rst:60]: Perhaps move the references to the typical files to the bottom to easy copy/paste?

**Pull request merged**